File: python/p001.py
def NoM(b, k):
    """Number of multiples of k in [1, b)"""
    return (b - 1) // k


def Sn(a1, an, n):
    """Sum of arithmetic series"""
    return (a1 + an) * n // 2


def BM(n, k):
    """Biggest multiple of k below n"""
    return (n - 1) - (n - 1) % k


A = Sn(3, BM(1000, 3), NoM(1000, 3))
B = Sn(5, BM(1000, 5), NoM(1000, 5))
C = Sn(15, BM(1000, 15), NoM(1000, 15))

print(A + B - C)

File: typescript/src/p001.ts
const MAX = 1000;

/** Number of multiples of k in [1, m) */
const NoM = (k: number, m: number = MAX) => {
  return Math.floor((m - 1) / k);
};

/** Sum of an arithmetic series */
const Sn = (a1: number, an: number, n: number) => {
  return ((a1 + an) / 2) * n;
};

/** Biggest multiple of k below m */
const BM = (k: number, m: number = MAX) => {
  return m - 1 - ((m - 1) % k);
};

const A = Sn(3, BM(3), NoM(3));
const B = Sn(5, BM(5), NoM(5));
const C = Sn(15, BM(15), NoM(15));

console.log(A + B - C);
